src/fairseq2/optim/lr_scheduler/factory.py

```diff
@@ -85,14 +91,42 @@ def create_cosine_annealing_lr(
     else:
         cycle_len = config.cycle_len
 
+    final_lr: float = 0.0  # using dummy value to make mypy happy
+
+    # set final lr based on config
+    if config.final_lr_scale is not None:
+        final_lr = optimizer.param_groups[0]["lr"] * config.final_lr_scale
+        assert (
```

Comment:
Do you mind converting these `assert` statements to `ValueError`s? So instead this, writing:
```
if config.final_lr is not None:
    raise ValueError("....")
```